gradio/helpers.py

```diff
@@ -290,13 +272,38 @@ def __init__(
         self.cached_indices_file = Path(self.cached_folder) / "indices.csv"
         self.run_on_click = run_on_click
         self.cache_event: Dependency | None = None
+        self.non_none_processed_examples = UnhashableKeyDict()
+
+        for example in examples:
+            self._get_processed_example(example)
+
+    def _get_processed_example(self, example):
```

Comment:
Once an example is processed, it is stored to the "self.non_none_processed_examples" dictionary, because processing an example can take time and we don't want a user to experience lag every time they click on an example.